### ISSUE TITLE
random_polyagamma sometimes hangs in version 1.3.1

### ISSUE THREAD

**Issue by macks22** [zoj613/polyagamma#102]
## Description of issue

The `random_polyagamma` function hangs without accepting a `KeyboardInterrupt` with some inputs in version 1.3.1.

### Environment setup

On both OSX and Linux:
```
conda create -y -n pgsample -c conda-forge python=3.7 ipykernel pip
conda activate pgsample
pip install polyagamma==1.3.1
```

Run the following Python code, using the attached data. Execution hangs on the last line.
```python
import pickle
import numpy as np
import polyagamma


psi = np.loadtxt("psi.txt")
state_keys = np.loadtxt("state_keys.txt")
state = ("MT19937", state_keys, 566, 0, 0.0)

random_state = np.random.RandomState()
random_state.set_state(state)

assert np.isnan(psi).sum() == 0
assert np.isinf(psi).sum() == 0
print(np.min(psi), np.max(psi))

n = np.ones(len(psi), dtype='float64')
out = polyagamma.random_polyagamma(n, psi, random_state=random_state._bit_generator)
```
[state_keys.txt](https://github.com/zoj613/polyagamma/files/7077022/state_keys.txt)
[psi.txt](https://github.com/zoj613/polyagamma/files/7077023/psi.txt)

**Comment by zoj613** [comment]
Hi @macks22, I think this issue is the same as https://github.com/zoj613/polyagamma/issues/83 , which has been resolved by https://github.com/zoj613/polyagamma/pull/85. I cannot reproduce the example on 1.3.2-beta.3. I recommend you upgrade the version to the latest pre-release.

**Comment by zoj613** [comment]
Also, I would replace `n` with a `1` literal to save the extra step of creating an array, It will be broadcasted with psi's shape internally.
change:
```python
n = np.ones(len(psi), dtype='float64')
out = polyagamma.random_polyagamma(n, psi, random_state=random_state._bit_generator)
```
to
```python
out = polyagamma.random_polyagamma(1, psi, random_state=random_state._bit_generator)
```

**Comment by macks22** [comment]
@zoj613 thanks for the fast response and the tips! In `numpy==1.21.2`, there is no publicly available `bit_generator` attribute on the `numpy.random.RandomState` object. I'm not sure when this was removed.

Is there a `polyagamma` version available via `conda` that has this fix incorporated? We are using `polyagamma` as part of a package which has the requirement that it must be `conda`-installable. This is a requirement for us in order to compatible with another package whose users are restricted to the use of `conda` for installation. Hence all of our dependencies must be `conda`-installable.

**Comment by zoj613** [comment]
> @zoj613 thanks for the fast response and the tips! In `numpy==1.21.2`, there is no publicly available `bit_generator` attribute on the `numpy.random.RandomState` object. I'm not sure when this was removed.
> 
> Is there a `polyagamma` version available via `conda` that has this fix incorporated? We are using `polyagamma` as part of a package which has the requirement that it must be `conda`-installable. This is a requirement for us in order to compatible with another package whose users are restricted to the use of `conda` for installation. Hence all of our dependencies must be `conda`-installable.

Oh yeah, that was my mistake. The public `bit_generator` is only available for the `np.random.Generator` class. The version on conda is `v1.3.2b3`, so it should satisfy your requirements.

**Comment by macks22** [comment]
I've confirmed that this bug is resolved with the `1.3.2b3` version of `polyagamma`. Thanks @zoj613!
